src/main/java/uk/gov/hmcts/ccd/domain/service/getevents/AuthorisedGetEventsOperation.java

```diff
@@ -74,12 +85,19 @@ public AuthorisedGetEventsOperation(@Qualifier("classified") GetEventsOperation
             throw new ValidationException("Cannot find case type definition for  " + caseTypeId);
         }
 
-        Set<String> userRoles = userRepository.getUserRoles();
-        if (userRoles == null || userRoles.isEmpty()) {
+        if (caseUserRoles == null || caseUserRoles.isEmpty()) {
             throw new ValidationException("Cannot find user roles for the user");
         }
 
-        return verifyReadAccess(events, userRoles, caseType);
+        return verifyReadAccess(events, caseUserRoles, caseType);
+    }
+
+    private Set<String> getUserRoles(String caseId) {
```

Comment:
can you rename this to `getAccessRoles` since these are no longer only user roles but it's now  a union of user roles and case roles

Reply:
we should also move this method to `CaseAccessService` to prevent duplicating the same logic in the future